### ISSUE TITLE
Cannot compile PDB on Mono - CS0041: Unexpected error writing debug information -- 'The method or operation is not implemented.'

### ISSUE THREAD

**Issue by MAX-POLKOVNIK** [dotnet/roslyn#8217]
There is sample: 
```
using System;
using System.IO;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;

namespace TestR
{
	class MainClass
	{
		public static void Main (string[] args)
		{
			var code = @"						 
						namespace TestR
						{
							class MainClass
							{
								public static void AAA (string[] args)
								{
								}
							}
						}
						";
            
			var syntaxTree = CSharpSyntaxTree.ParseText(code);
            
			var references = new [] {
				MetadataReference.CreateFromFile (typeof(Type).Assembly.Location)
			};
            
			var compilation = CSharpCompilation.Create(
				"assembly",
				new[] {syntaxTree},
				references,
				new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));


			using (var ms = new MemoryStream ())
			using (var pdbs = new MemoryStream ()) 
			{
				var result = compilation.Emit (ms, pdbs);

				if (!result.Success) 
				{
					throw new Exception ("Not compiled: " + result.Diagnostics[0]);
				}

				Console.WriteLine ("Compiled and saved");
			}
		}
	}
}
```

But if I remove pdbs from **compilation.Emit (ms, pdbs)** that will compile successfully.
Running without mono all works.

**Comment by tmat** [comment]
What OS are you running on?

**Comment by tmat** [comment]
Also, could you run under the debugger and share a stack trace of the first-chance exception?

**Comment by MAX-POLKOVNIK** [comment]
Running on OS X 10.11.1.
Roslyn do not throw any exception. I throw this manually.
`throw new Exception ("Not compiled: " + result.Diagnostics[0]);`

**Comment by tmat** [comment]
By default Roslyn produces Windows-specific PDBs to ensure backward compatibility. These PDBs enable the full feature set of Windows debuggers (such as expression evaluation and Edit and Continue). However, these PDBs are not possible to read or write on non-Windows systems.

To generate cross-platform Portable PDB format you can set DebugInformationFormat emit option
http://source.roslyn.io/#Microsoft.CodeAnalysis/Emit/EmitOptions.cs,61
to PortablePdb.

Note that this format is new and there is currently a limited support for it. I believe the latest Mono build runtime supports it.

**Comment by isomorphisms** [comment]
I am getting this error too.

**Comment by isomorphisms** [comment]
I am getting this error too.

**Comment by ramzzzay** [comment]
OS X 10.10.1

uname -a
Darwin iMac-Anton.local 14.0.0 Darwin Kernel Version 14.0.0: Fri Sep 19 00:26:44 PDT 2014; root:xnu-2782.1.97~2/RELEASE_X86_64 x86_64

mono --version
Mono JIT compiler version 4.4.0 (mono-4.4.0-branch-c7-baseline/5995f74 Thu Jun  2 15:13:10 EDT 2016)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  x86
	Disabled:      none
	Misc:          softdebug
	LLVM:          yes(3.6.0svn-mono-master/a173357)
	GC:            sgen

asp.net mvc5&web api2
Error on build:
Error:CS0041: Unexpected error writing debug information -- 'The method or operation is not implemented.'

![screen shot 2016-06-22 at 22 20 56](https://cloud.githubusercontent.com/assets/5748239/16280320/9d03598c-38c7-11e6-995a-60da23d9a0e0.png)

**Comment by golfguy0082** [comment]
I'm sorry to be an idiot, but where do you set the DebugInformationFormat option?  Is this in the csproj somewhere?

**Comment by hanssens** [comment]
@golfguy0082 I had the same problem with a .NET 4.6 solution compiling on macOS. For this, I found the [document "Portable PDB's"](https://github.com/OmniSharp/omnisharp-vscode/wiki/Portable-PDBs#how-to-generate-portable-pdbs) in the OmniSharp repo which explains this a bit.

Basically put, you can set this in the .csproj file:

```
<DebugType>portable</DebugType>
```

Or, it if's a dotnet core project, set it in the project.json:

```
"buildOptions": {
    "debugType": "portable"
}
```

portablepdb's seems to be a new thing introduced with dotnet core, so results may very depending on your IDE.

**Comment by golfguy0082** [comment]
Thank you very much @hanssens!  Very helpful!

**Comment by dharanesh03** [comment]
@hanssens I did that and i ended up with this error.
Error CS1902: Invalid option 'portable' for /debug; must be full or pdbonly (CS1902)"

Any thoughts on this?

**Comment by hanssens** [comment]
@dharanesh03 no, unfortunately not. It seems that the compiler, or IDE, you're using doesn't support the "portable" option just yet. As mentioned in the Omnisharp docs this is pretty new.

**Comment by castrojr913** [comment]
@dharanesh03 @hanssens the same thing :(. I couldn't compile the project

**Comment by tmat** [comment]
The original issue is about using Roslyn emit APIs to produce a binary and a PDB on OS X. The answer to that is:

> To generate cross-platform Portable PDB format you can set DebugInformationFormat emit option http://source.roslyn.io/#Microsoft.CodeAnalysis/Emit/EmitOptions.cs,61 to PortablePdb.

That's related but different issue from not being able to build a .csproj on OS X/Linux. 
@castrojr913 @dharanesh03 What is the version of the compiler and IDE you're using?

**Comment by castrojr913** [comment]
@tmat IDE: Visual Studio Preview 4 (7.0 Build 1566). OS: Mac OSX 10.12.3 (Sierra).

**Comment by tmat** [comment]
@castrojr913 Does setting debug type in .csproj fix the issue?

```
<DebugType>portable</DebugType>
```

**Comment by ncapito** [comment]
Didn't fix it for me.  Same error.

**Comment by Eric0625** [comment]
> @castrojr913 Does setting debug type in .csproj fix the issue?
> 
> ```
> <DebugType>portable</DebugType>
> ```

It's 2020 now and this type is still not supported by the latest version of VS for MAC
